### PR TITLE
Refactor EcdhPublicKey and EcdhPrivateKey Class

### PR DESCRIPTION
**Pull request by HamdaanAliQuatil**

### CODE REVIEW

**Comment by HamdaanAliQuatil** [comment]
Part of #105

File: lib/src/impl_interface/impl_interface.dart [hunk 1]
```diff
@@ -17,9 +17,12 @@ library impl_stub;
 import 'dart:typed_data';
 import 'dart:async';
 
+import '../webcrypto/webcrypto.dart';
```
Comment (jonasfj) [inline lib/src/impl_interface/impl_interface.dart:20]: What we don't want this. That creates cyclic imports. It becomes messy, let's avoid that.

We can just make a `KeyPairImpl`, or fully move `KeyPair` into `impl_interface.dart`.

Comment (HamdaanAliQuatil) [reply lib/src/impl_interface/impl_interface.dart:20]: Yes, this makes sense. tysm. modifying accordingly

File: lib/src/impl_interface/impl_interface.ecdh.dart [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+part of 'impl_interface.dart';
+
+abstract interface class StaticEcdhPrivateKeyImpl {
+  Future<EcdhPrivateKeyImpl> importPkcs8Key(List<int> keyData, EllipticCurve curve);
+  Future<EcdhPrivateKeyImpl> importJsonWebKey(Map<String, dynamic> jwk, EllipticCurve curve);
+  Future<KeyPair<EcdhPrivateKeyImpl, EcdhPublicKeyImpl>> generateKey(EllipticCurve curve);
```
Comment (jonasfj) [inline lib/src/impl_interface/impl_interface.ecdh.dart:20]: We could do this right?
```suggestion
  Future<(EcdhPrivateKeyImpl, EcdhPublicKeyImpl)> generateKey(EllipticCurve curve);
```

File: lib/src/impl_interface/impl_interface.dart (hunk 1 above)
Comment (jonasfj) [reply lib/src/impl_interface/impl_interface.dart:20]: Or we can just do as proposed in:
https://github.com/google/webcrypto.dart/pull/145/files#r1804350232

Specifcally, we do:
```
Future<(EcdhPrivateKeyImpl, EcdhPublicKeyImpl)> generateKey(EllipticCurve curve);
```

Then we can still make it into a `KeyPair` in `webcrypto/webcrypto.dart` where we have to wrap `EcdhPrivateKeyImpl` into `EcdhPrivateKey`.

**Comment by jonasfj** [comment]
hmm, not sure my merge was correct, I might have created more problems than I solved :see_no_evil:

**Review by jonasfj — APPROVED** [review]

**Pull request merged**